File: services/language-service/src/main/java/dev/animedia/languageservice/application/service/UpdateLanguageService.java
package dev.animedia.languageservice.application.service;

import dev.animedia.languageservice.application.dto.LanguageDto;
import dev.animedia.languageservice.application.mapper.LanguageApplicationMapper;
import dev.animedia.languageservice.application.usecase.UpdateLanguageUseCase;
import dev.animedia.languageservice.domain.exception.LanguageNotFoundException;
import dev.animedia.languageservice.domain.model.Language;
import dev.animedia.languageservice.domain.repository.LanguageCommandRepository;
import dev.animedia.languageservice.domain.repository.LanguageQueryRepository;
import jakarta.transaction.Transactional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UpdateLanguageService implements UpdateLanguageUseCase {
    private final LanguageCommandRepository languageCommandRepository;
    private final LanguageQueryRepository languageQueryRepository;
    private final LanguageApplicationMapper languageApplicationMapper;

    @Autowired
    public UpdateLanguageService(
        LanguageCommandRepository languageCommandRepository,
        LanguageQueryRepository languageQueryRepository,
        LanguageApplicationMapper languageApplicationMapper
    ) {
        this.languageCommandRepository = languageCommandRepository;
        this.languageQueryRepository = languageQueryRepository;
        this.languageApplicationMapper = languageApplicationMapper;
    }

    @Transactional
    @Override
    public LanguageDto update(LanguageDto request) {
        Language language = languageQueryRepository.findByCode(request.code())
            .orElseThrow(LanguageNotFoundException::new);

        language.update(request.name(), request.isActive(), request.isDefault(), request.sortOrder(), request.flagEmoji());

        Language updated = languageCommandRepository.update(language);

        return languageApplicationMapper.toDto(updated);
    }
}

File: services/social-service/src/core/Application/Language/Service/Private/UpdateLanguageService.php
<?php

declare(strict_types=1);

namespace Core\Application\Language\Service\Private;

use Core\Application\Language\DTO\Private\LanguageDto;
use Core\Application\Language\Exception\LanguageNotFoundException;
use Core\Application\Language\Mapper\LanguageApplicationMapperInterface;
use Core\Application\Language\UseCase\Private\UpdateLanguageUseCase;
use Core\Domain\Language\Repository\LanguageCommandRepositoryInterface;
use Core\Domain\Language\Repository\LanguageQueryRepositoryInterface;

final readonly class UpdateLanguageService implements UpdateLanguageUseCase
{
    public function __construct(
        private LanguageQueryRepositoryInterface $languageQueryRepository,
        private LanguageCommandRepositoryInterface $languageCommandRepository,
        private LanguageApplicationMapperInterface $languageApplicationMapper
    ) {}

    #[\Override]
    public function execute(LanguageDto $languageDto): LanguageDto
    {
        $language = $this->languageQueryRepository->findByCode($languageDto->languageIsoCode);
        if ($language === null) {
            throw new LanguageNotFoundException($languageDto->languageIsoCode);
        }

        $language->update(
            name: $languageDto->name,
            active: $languageDto->isActive
        );

        $updated = $this->languageCommandRepository->update($language);

        return $this->languageApplicationMapper->toPrivateLanguageDto($updated);
    }
}
